autoprecompiles/src/optimizer_documentation.md

```diff
@@ -204,6 +204,63 @@ TODO Continue with the abstraction using Range Constraints.
 ## Combining Range Constraints
 
 
+## Grouped Expressions
+
+The main data structure used for algebraic expressions the _Grouped Expression_.
+A Grouped Expression consists of a constant term, a list of linear terms (a list of pairs of a non-zero coefficient
+and a variable) and a list of quadratic terms (a list of pairs of Grouped Expressions).
+
+For affine Algebraic Expressions, the Grouped Expression provides a normal form (if sorted by variable).
+It even provides a normal form for affine Algebraic Constraints if multiplied by a factor such that
```

Comment:
Why Algebraic Constraints in capital initials?

Reply:
I'm writing all properly defined concepts in capitals, and in italics (using `_`...`_`) at the place where they are defined.